python/packages/nisar/workflows/stage_watermask.py

```diff
@@ -289,6 +291,20 @@ def translate_watermask(vrt_filename, outpath, x_min, x_max, y_min, y_max):
     input_x_min, xres, _, input_y_max, _, yres = ds.GetGeoTransform()
     length = ds.GetRasterBand(1).YSize
     width = ds.GetRasterBand(1).XSize
+
+    # Declare lambda function to snap min/max X and Y
+    # coordinates over the water grid
+    snap_coord = lambda val, snap, offset, round_func: round_func(  # noqa: E731
+        float(val - offset) / snap) * snap + offset
```

Comment:
Instead of adding the `# noqa: E731` comment to disable the [E731](https://docs.astral.sh/ruff/rules/lambda-assignment/) check, we could just implement its suggestion :shrug:. It has the extra benefit of also being more readable (IMO).

```suggestion
    def snap_coord(val, snap, offset, round_func):
        return round_func(float(val - offset) / snap) * snap + offset
```